JortPob/IconManager.cs

```diff
@@ -0,0 +1,505 @@
+            Lort.NewTask("Binding Sheets", 2);
+            AddSheets(hiLayoutPath, hiTpfPath, "Hi"); Lort.TaskIterate();
+            AddSheets(lowLayoutPath, lowTpfPath, "Low"); Lort.TaskIterate();
+
+            /* Clean up */
+            foreach ((Layout layout, Bitmap bitmap) tuple in sheets) { tuple.bitmap.Dispose(); }
+            foreach ((IconInfo iconInfo, Bitmap bitmap) tuple in bitmaps) { tuple.bitmap.Dispose(); }
+
+            /* Do large icons for use in the description area */
+            const string hiPath = @"menu\hi\00_solo";
+            const string lowPath = @"menu\low\00_solo";
+
+            Lort.Log($"Generating {icons.Count()} previews...", Lort.Type.Main);
+            Lort.NewTask("Generating Previews", icons.Count());
+
+            void AddIcons(string path)
+            {
+                /* Load BXF4 from elden ring directory (requires game unpacked!) */
+                BXF4 bxf = BXF4.Read($"{Const.ELDEN_PATH}Game\\{path}.tpfbhd", $"{Const.ELDEN_PATH}Game\\{path}.tpfbdt");
+                List<BinderFile> filesToInsert = new();
+
+                /* Generate binder files to add from dds texture files */
+                foreach(IconInfo icon in icons)
+                {
+                    byte[] data = System.IO.File.ReadAllBytes($"{Const.MORROWIND_PATH}Data Files\\icons\\{icon.path}");
+
+                    Bitmap bitmap = Common.DDS.DDStoBitmap(data);
+                    Bitmap scaledBitmap = Common.Utility.XbrzUpscale(bitmap, 6); // 32x32x -> 192x192
+                    Bitmap linearScaledBitmap = Common.Utility.LinearToSRGB(scaledBitmap);
+                    byte[] scaledDDS = Common.DDS.BitmapToDDS(linearScaledBitmap, DXGI_FORMAT.BC2_UNORM);
+                    scaledBitmap.Dispose();
+                    linearScaledBitmap.Dispose();
+
+                    int format = JortPob.Common.DDS.GetTpfFormatFromDdsBytes(scaledDDS);
+
+                    TPF tpf = new TPF();
+                    tpf.Encoding = 1;
+                    tpf.Flag2 = 3;
+                    tpf.Platform = TPF.TPFPlatform.PC;
+                    tpf.Compression = DCX.Type.DCX_KRAK;
+
+                    TPF.Texture tex = new(icon.TextureName(), (byte)format, 0, scaledDDS, TPF.TPFPlatform.PC);
+                    tpf.Textures.Add(tex);
+
+                    BinderFile bf = new();
+                    bf.Name = icon.FileName();
+                    bf.Bytes = tpf.Write();
+                    bf.ID = bxf.Files.Count();
+
+                    filesToInsert.Add(bf);
+                }
+
+                /* Reindex after inserting new files */
+                int insertAt = 0;
+                for(int i= 0;i<bxf.Files.Count();i++)
+                {
+                    BinderFile bf = bxf.Files[i];
+                    if(bf.Name.StartsWith("00_Solo\\MENU_Knowledge_"))
+                    {
+                        insertAt = i;
+                    }
+                }
```

Comment:
this is currently finding the _last_ index to replace at. assuming we do only want that last index, i think we can use [FindLastIndex](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.findlastindex?view=net-10.0)
```c#
int insertAt = bxf.Files.FindLastIndex(bf => bf.Name.StartsWith("00_Solo\\MENU_Knowledge_"));
if (insertAt != -1) {
    bxf.Files.InsertRange(insertAt + 1, filesToInsert);
}
```